MetaMask/metamask-extension
# feat: add privacy mode

## **Description**

Adds a privacy mode toggle (an eye icon next to the main balance) that hides all sensitive information/token balances

**UPDATE**
Here is feedback from @amandaye0h and has been currently implemented in this PR
[Figma](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?node-id=6219-62460&t=aeTv5cenoUPUrg1c-4)

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/28021?quickstart=1)

## **Related issues**

Fixes:
https://github.com/MetaMask/MetaMask-planning/issues/3416
https://github.com/MetaMask/MetaMask-planning/issues/3418
https://github.com/MetaMask/MetaMask-planning/issues/3419

## **Manual testing steps**

1. Go to the Wallet page
2. Click on the new Eye icon next to the balance
3. All balances should be hidden
4. Click on the Eye icon once again
5. All balances should be shown

## **Screenshots/Recordings**



### **Before**



### **After**

https://github.com/user-attachments/assets/2950ac0c-593d-4daa-aa5d-3e6c3a2d5598

https://github.com/user-attachments/assets/6371c2a2-04fa-48a3-8744-991a1540d5f2

<img width="496" alt="Screenshot 2024-10-22 at 18 43 19" src="https://github.com/user-attachments/assets/d7c2f681-75c7-4be0-921b-f3c5186f8d4a">



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
